core/shared/utils/bh_time.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include "bh_time.h"
+
+uint64
+convert_timespec(const struct timespec *ts)
+{
+    if (ts->tv_sec < 0)
+        return 0;
+    if ((uint64)ts->tv_sec >= UINT64_MAX / 1000000000)
+        return UINT64_MAX;
+    return (uint64)ts->tv_sec * 1000000000 + (uint64)ts->tv_nsec;
+}
+
+static bool
+convert_clockid(bh_clock_id_t in, _clockid_t *out)
```

Comment:
This function is posix-specific so shouldn't be in the common code. I'd suggest we move it to posix_time.c

Reply:
Also, this function should use `clockid_t` type:
```suggestion
convert_clockid(bh_clock_id_t in, clockid_t *out)
```